backend/btrixcloud/basecrawls.py

```diff
@@ -688,6 +733,30 @@ async def update_crawl(
     @app.post("/orgs/{oid}/all-crawls/delete", tags=["all-crawls"])
     async def delete_crawls_all_types(
         delete_list: DeleteCrawlList,
+        user: User = Depends(user_dep),
         org: Organization = Depends(org_crawl_dep),
     ):
-        return await ops.delete_crawls_all_types(delete_list, org)
+        crawls: list[str] = []
+        uploads: list[str] = []
+
+        for crawl_id in delete_list.crawl_ids:
+            crawl = await ops.get_crawl_raw(crawl_id, org)
+
+            if (crawl.get("userid") != user.id) and not org.is_owner(user):
+                raise HTTPException(status_code=403, detail="not_allowed")
+
+            type_ = crawl.get("type")
+            if type_ == "crawl":
+                crawls.append(crawl_id)
+                if not crawl.get("finished"):
+                    try:
+                        await ops.shutdown_crawl(crawl_id, org, graceful=False)
```

Comment:
We should probably move this into the shared delete_crawls() method so it is applied even when deleting just crawls.